### PR TITLE
[Bug] Fix ColumnIdentifier to string to let dictionary generate error information understandable.

### PR DESCRIPTION
**Pull request by Zhangshunyu**
Fix ColumnIdentifier to string to let dictionary generate error information understandable.

### CODE REVIEW

**Comment by Zhangshunyu** [comment]
@QiangCai pls review this code, thank you!

File: core/src/main/java/org/carbondata/core/carbon/ColumnIdentifier.java [hunk 1]
```diff
@@ -106,4 +106,8 @@ public DataType getDataType() {
     return true;
   }
 
+  @Override public String toString() {
+    return "ColumnIdentifier{" + "columnId='" + columnId + '\'' + '}';
```
Comment (QiangCai) [inline core/src/main/java/org/carbondata/core/carbon/ColumnIdentifier.java]: suggest to use '[]' instead of '{}'

**Comment by Zhangshunyu** [comment]
@QiangCai Done, please check again.

**Comment by kumarvishal09** [comment]
@Zhangshunyu can u please add more detail about the issue.I think adding column name will be more useful as columnid is random string

File: core/src/main/java/org/carbondata/core/carbon/ColumnIdentifier.java [hunk 2]
```diff
@@ -106,4 +106,8 @@ public DataType getDataType() {
     return true;
   }
 
+  @Override public String toString() {
+    return "ColumnIdentifier[" + "columnId='" + columnId + '\'' + ']';
```
Comment (QiangCai) [inline core/src/main/java/org/carbondata/core/carbon/ColumnIdentifier.java]: return "ColumnIdentifier [columnId=" + columnId + "]";

**Comment by Zhangshunyu** [comment]
because there is no column name in ColumnIdentifier and we can see the column id and column name in shcema file. But if we modify the  ColumnIdentifier  class only for this log info, it will affect many places, so we think it is no need to add the column name in ColumnIdentifier  since we can recognize it from the schema file.

**Comment by kumarvishal09** [comment]
Ok. But logging column id wont be useful as user doesn't know about the column id and for the developers it wont be simple to sync columnid with the column name. Please correct me if I am worng

**Comment by QiangCai** [comment]
@kumarvishal09  
We can open schema file directly to find column id and column name. It is easy to map column id with  column name.
After checked code, We found that adding column name for ColumnIdentifier is more difficult than adding table name for CarbonTableIdentifier. I think We need your help.

**Comment by kumarvishal09** [comment]
Ok @QiangCai merge this PR.

**Comment by QiangCai** [comment]
LGTM

**Pull request merged**